scripts/contributor_dashboard_debug_test.py

```diff
@@ -0,0 +1,286 @@
+
+from scripts import contributor_dashboard_debug
+
+import firebase_admin
+from firebase_admin import auth as firebase_auth
+
+import requests
+from typing import Any, Dict, List
+import webapp2
+
+auth_services = models.Registry.import_auth_services()
+
+
+class MockResponse:
+
+    def __init__(
+        self,
+        ok: bool = True,
+        json: Dict[str, Any] | None = None,
+        status_code: int = 200,
+        reason: str = 'foo'
+    ) -> None:
+        if json is None:
+            json = {}
+        self.ok = ok
+        self.json_dict = json
+        self.status_code = status_code
+        self.reason = reason
+
+    def json(self) -> Dict[str, Any]:
+        """Get json dict or raise ValueError if json_dict not a dict."""
+        if not isinstance(self.json_dict, dict):
+            raise ValueError('Payload not JSON.')
+        return self.json_dict
+
+
+class ContributorDashboardDebugInitializerTests(test_utils.GenericTestBase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.firebase_sdk_stub = (
+            firebase_auth_services_test.FirebaseAdminSdkStub())
+        self.firebase_sdk_stub.install(self)
+        self.token_by_email: Dict[str, str] = {}
+        self.token_of_current_user: str | None = None
+        self.initializer = (
+            contributor_dashboard_debug.ContributorDashboardDebugInitializer(
+                base_url=''))
+
+        self.initializer.csrf_token = self.get_new_csrf_token()
+
+        self.request_swap = self.swap(
+            self.initializer.session,
+            'request',
+            self.mock_request)
+
+    def tearDown(self) -> None:
+        self.firebase_sdk_stub.uninstall()
+        self.logout()
+        super().tearDown()
+
+    def mock_request(
+        self,
+        method: str,
+        url: str,
+        params: Dict[str, Any] | None = None,
+        headers: Dict[str, Any] | None = None
+    ) -> Any:
+        """Returns a mock response for the given request."""
+        if method == 'GET':
+            with self.swap(
+                base, 'load_template', test_utils.mock_load_template):
+                return self.testapp.get(url, params=params, headers=headers)
+        if method == 'POST':
+            return self.testapp.post(url, params=params, headers=headers)
+        if method == 'PUT':
+            return self.testapp.put(url, params=params, headers=headers)
+
+    def test_populate_debug_data(self) -> None:
+        init_app_swap = self.swap_with_call_counter(
+            firebase_admin, 'initialize_app')
+        create_user_swap = self.swap(
+            firebase_auth, 'create_user', self.mock_firebase_auth_create_user)
+        begin_session_swap = self.swap(
+            self.initializer, '_sign_in',
+            self.mock_login_as_admin)
+
+        with self.request_swap, create_user_swap, begin_session_swap, (
+            init_app_swap) as init_app_counter:
+            self.initializer.populate_debug_data()
```

Comment:
This indentation is confusing. Maybe:

```suggestion
        with self.request_swap, create_user_swap, begin_session_swap:
            with init_app_swap as init_app_counter:
                self.initializer.populate_debug_data()
```

ditto elsewhere in this file

Reply:
Thanks. That's a good suggestion.